packages/client/src/object/Attachment.ts

```diff
@@ -14,12 +14,23 @@
  * limitations under the License.
  */
 
-import type { Attachment as IAttachment } from "@osdk/client.api";
+import type {
+  Attachment as IAttachment,
+  AttachmentUpload as IAttachmentUpload,
+} from "@osdk/client.api";
 
 export class Attachment implements IAttachment {
```

Comment:
Yea I was trying to get more SA on why we did this as well, found this old PR you discussed this on: https://github.com/palantir/osdk-ts/pull/21. I just followed the pattern of what we had so far for the additions I made, it seems like it was mainly for the data conversions because just checking for `{rid: string}` was risky